File: src/main/native/src/stop_watch.cpp
#include "stop_watch.h"


namespace graph {


long TimeNanosecondsNow() {
        auto fromEpoch = std::chrono::high_resolution_clock::now().time_since_epoch();
        return std::chrono::duration<long, std::nano>(fromEpoch).count();
}


StopWatch::StopWatch() { reset(); }

void StopWatch::setStartTick() { startTick = TimeNanosecondsNow(); }

void StopWatch::setStopTick() {
        stopTick = TimeNanosecondsNow();
        accTime += (stopTick - startTick);
}

long StopWatch::delta() { return stopTick - startTick; }

void StopWatch::reset() {
        startTick = 0L;
        stopTick = 0L;
        accTime = 0L;
}

} // namespace graph

File: src/main/java/com/mstruzek/msketch/solver/StopWatch.java
package com.mstruzek.msketch.solver;

import java.util.function.LongSupplier;

public class StopWatch {
    private static final LongSupplier nanoClock = System::currentTimeMillis;
    long startTick;
    long stopTick;
    long accTime;

    StopWatch() {
    }

    public void startTick() {
        this.startTick = nanoClock.getAsLong();
    }

    public void stopTick() {
        this.stopTick = nanoClock.getAsLong();
        this.accTime += (stopTick - startTick);
    }

    public long delta() {
        return stopTick - startTick;
    }

    public void reset(){
        this.stopTick = 0;
        this.startTick = 0;
        this.accTime = 0;

    }
}
